### PR TITLE
[SPACES] Fixes in copy and move operations

### PR DESCRIPTION
**Pull request by JuancaG05**
## Related Issues
App: https://github.com/owncloud/android/issues/3863

Library PR: https://github.com/owncloud/android-library/pull/543
_____

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/ui/activity/FolderPickerActivity.kt [hunk 1]
```diff
@@ -158,8 +183,8 @@ open class FolderPickerActivity : FileActivity(),
         }
         // If current file is root folder
         else if (currentDirDisplayed.parentId == OCFile.ROOT_PARENT_ID) {
-            // If we are not in COPY mode, close the activity
-            if (pickerMode != PickerMode.COPY) {
+            // If we are not in COPY mode, or if we are in COPY mode and spaces are not allowed, close the activity
+            if (pickerMode != PickerMode.COPY || (pickerMode == PickerMode.COPY && !AccountUtils.isSpacesFeatureAllowedForAccount(baseContext, account, capabilities))) {
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/activity/FolderPickerActivity.kt:187]: Same here, we can rely on the `spaceId` existence. What do you think?

```suggestion
            if (pickerMode != PickerMode.COPY || (pickerMode == PickerMode.COPY && currentDirDisplayed.spaceId == null)) {
```

**Review by abelgardep — CHANGES_REQUESTED** [review]
@JuancaG05 Last changes and ready to go 👍

File: owncloudApp/src/main/java/com/owncloud/android/ui/activity/FolderPickerActivity.kt [hunk 2]
```diff
@@ -72,8 +86,19 @@ open class FolderPickerActivity : FileActivity(),
...
+                            capabilities = uiResult.data
+                            if (AccountUtils.isSpacesFeatureAllowedForAccount(baseContext, account, capabilities)) {
+                                // Show the list of spaces
+                                initAndShowListOfSpaces()
+                            } else {
+                                // Show the personal space
+                                initAndShowListOfFilesFragment(spaceId = null)
+                            }
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (JuancaG05) [reply owncloudApp/src/main/java/com/owncloud/android/ui/activity/FolderPickerActivity.kt:101]: 👍

File: owncloudApp/src/main/java/com/owncloud/android/ui/activity/FolderPickerActivity.kt (hunk 1 above)
Comment (JuancaG05) [reply owncloudApp/src/main/java/com/owncloud/android/ui/activity/FolderPickerActivity.kt:187]: 💯

**Review by abelgardep — APPROVED** [review]

**Pull request merged**